drivers/hmis/app/graphql/mutations/ce/start_ce_referral_step.rb

```diff
@@ -20,8 +20,17 @@ def resolve(referral_id:, step_id:)
       referral.opportunity.with_lock do
         engine = referral.workflow_engine
         step = engine.active_steps.find(step_id)
+
+        participants = referral.participants_for_swimlane(step.swimlane.id)
+        raise 'Cannot start a referral step with no participants' unless participants.any?
+
+        # TODO(#7395): permission
+
+        step.assignments.find_or_create_by!(user: current_user)
```

Comment:
Let's discuss: I added step assignment here; Should this be moved into the engine logic? I kept it out here since I wasn't totally sure it is generalized behavior that we always want on any type of workflow engine.

Reply:
Yes I think the logic for filling that column should be internal to the engine. It is a general `wfe_` table, after all. I can't think of a use case for exposing that logic to allow the caller to change _when_ the step is assigned (or who its assigned to?) which I think is what you're getting at. Lets err on the side of keeping it encapsulated in the engine. (update this so `engine.start_step!(step, user: current_user)` populates the assignment table instead. )